stackslib/src/net/relay.rs

```diff
@@ -528,6 +529,41 @@ impl Relayer {
         Ok(())
     }
 
+    /// Given Nakamoto blocks pushed to us, verify that they correspond to expected block data.
+    pub fn validate_nakamoto_blocks_push(
+        conn: &SortitionDBConn,
+        blocks_data: &NakamotoBlocksData,
+    ) -> Result<(), net_error> {
+        for NakamotoBlocksDatum(consensus_hash, block) in blocks_data.blocks.iter() {
+            let block_hash = block.header.block_hash();
+
+            // is this the right Stacks block for this sortition?
+            let sn = match SortitionDB::get_block_snapshot_consensus(conn.conn(), consensus_hash)? {
+                Some(sn) => {
+                    if !sn.pox_valid {
+                        info!( "Pushed block from consensus hash {consensus_hash} corresponds to invalid PoX state",);
+                        continue;
+                    }
+                    sn
+                }
+                None => {
+                    // don't know about this yet
+                    continue;
+                }
+            };
+
+            if !sn.sortition || sn.winning_stacks_block_hash != block_hash {
```

Comment:
Is this check valid in Nakamoto? As currently implemented in this PR, it's okay, because `consensus_hash` is separated from the `block`, but this PR doesn't make clear that this was actually the intention. I think this needs to be made more explicit.

In a little more detail: `winning_stacks_block_hash` in Nakamoto will correspond to a Stacks block previously mined -- so its block header's `consensus_hash` field points at a different bitcoin block. Instead, it needs to be looked up by the `announcement_ch` (or the like: the consensus hash in which the winning miner announced the block hash).